doc/user/methods/random_forest.md

```diff
@@ -96,6 +96,8 @@ std::cout << arma::accu(predictions == 3) << " test points classified as class "
 | `minLeafSize` | `size_t` | Minimum number of points in each leaf node of each decision tree. | `1` |
 | `minGainSplit` | `double` | Minimum gain for a node to split in each decision tree. | `1e-7` |
 | `maxDepth` | `size_t` | Maximum depth for each decision tree. (0 means no limit.) | `0` |
+| `dimSelector` | `DimensionSelectionType` | | `DimensionSelectionType()` |
+| `bootstrap` | `BootstrapType` | Bootstrap strategy. | `BootstrapType()` |    
```

Comment:
Let's not put this documentation here, where it might be confusing.  Instead let's add it as a bullet point introducing the constructor with all options at the beginning of the `Fully custom behavior` section, after we introduce all of the template parameters.